Sming/Services/SpifFS/spiffs_sming.c

```diff
@@ -110,35 +116,66 @@ static void spiffs_mount_internal(spiffs_config *cfg)
   cfg->hal_read_f = api_spiffs_read;
   cfg->hal_write_f = api_spiffs_write;
   cfg->hal_erase_f = api_spiffs_erase;
-  
-  uint32_t dat;
-  bool writeFirst = false;
-  flashmem_read(&dat, cfg->phys_addr, 4);
-  //debugf("%X", dat);
-
-  if (dat == UINT32_MAX)
-  {
-	  debugf("First init file system");
-	  spiffs_format_internal(cfg);
-	  writeFirst = true;
-  }
 
-  int res = SPIFFS_mount(&_filesystemStorageHandle,
-    cfg,
-    spiffs_work_buf,
-    spiffs_fds,
-    sizeof(spiffs_fds),
-    spiffs_cache,
-    sizeof(spiffs_cache),
-    NULL);
+  int res = spiffs_mount_minimal(cfg);
   debugf("mount res: %d\n", res);
 
-  if (writeFirst)
-  {
-	  file_t fd = SPIFFS_open(&_filesystemStorageHandle, "initialize_fs_header.dat", SPIFFS_CREAT | SPIFFS_TRUNC | SPIFFS_RDWR, 0);
-	  SPIFFS_write(&_filesystemStorageHandle, fd, (u8_t *)"1", 1);
-	  SPIFFS_fremove(&_filesystemStorageHandle, fd);
-	  SPIFFS_close(&_filesystemStorageHandle, fd);
+  if(res < 0) {
+	  int check = SPIFFS_check(&_filesystemStorageHandle);
+	  if(check < 0) {
+		  debugf("Unsuccessful SPIFFS check with error %d.\r\n", check);
```

Comment:
Change here `debugf` to `debug_w`. All debug messages that are showing up as a result on an error should use `debug_w`. 
Further all debug messages that should be informative and are understandable also from non-developers should use `debug_i`. And finally all debugging messages that should contain information about the inner workings of the system and are useful only to developers should use `debug_d`.